src/webui/www/private/scripts/mocha-init.js

```diff
@@ -1187,22 +1187,63 @@ const initializeWindows = () => {
 
     exportTorrentFN = async () => {
         const hashes = torrentsTable.selectedRowsIds();
-        for (const hash of hashes) {
+
+        if (hashes.length === 0)
+            return;
+
+        if (hashes.length === 1) {
+            // Single file - direct download
+            const hash = hashes[0];
             const row = torrentsTable.getRow(hash);
             if (!row)
-                continue;
+                return;
 
             const name = row.full_data.name;
             const url = new URL("api/v2/torrents/export", window.location);
-            url.search = new URLSearchParams({
-                hash: hash
+            url.search = new URLSearchParams({ hash });
+
+            await window.qBittorrent.Misc.downloadFile(
+                url,
+                `${name}.torrent`,
+                "QBT_TR(Unable to export torrent file)QBT_TR[CONTEXT=MainWindow]"
+            );
+        }
+        else {
+            // Multi-file - distributed in a zip
+            const zip = new JSZip();
+
+            const fetchTorrents = hashes.map(async (hash) => {
+                const row = torrentsTable.getRow(hash);
+                if (!row)
+                    return;
+
+                const name = row.full_data.name;
+                const url = new URL("api/v2/torrents/export", window.location);
+                url.search = new URLSearchParams({ hash });
+
+                try {
+                    const response = await fetch(url);
+                    if (!response.ok)
+                        throw new Error();
+
+                    const blob = await response.blob();
+                    zip.file(`${name}.torrent`, blob);
+                }
+                catch (error) {
+                    alert("QBT_TR(Unable to export torrent file)QBT_TR[CONTEXT=MainWindow]");
+                }
             });
 
-            // download response to file
-            await window.qBittorrent.Misc.downloadFile(url, `${name}.torrent`, "QBT_TR(Unable to export torrent file)QBT_TR[CONTEXT=MainWindow]");
+            // Wait for all fetches to finish
+            await Promise.all(fetchTorrents);
+
+            const content = await zip.generateAsync({ type: "blob" });
+            // Set output zip file date and time
+            const timestamp = new Date().toISOString().replace(/[:.]/g, "-").replace("T", "_").slice(0, 19);
```

Comment:
This will be at +0 timezone right? You live there? It will be bizarre for users living elsewhere.
I would suggest this instead: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toString